external-crates/move/crates/move-compiler/src/typing/expand.rs

```diff
@@ -368,6 +375,89 @@ fn inferred_numerical_value(
     }
 }
 
+fn inferred_string_value(
+    context: &mut Context,
+    _eloc: Loc,
+    value_loc: Loc,
+    value: Vec<u8>,
+    ty: &Type,
+) -> Option<T::Exp> {
+    use T::UnannotatedExp_ as E;
+    let diag_note =
+        || "String literals must be linked against the standard library to be constructed";
+
+    match &ty.value {
+        Type_::Apply(_, sp!(_, TypeName_::Builtin(sp!(_, bt))), args)
+            if bt.is_vector()
+                && args.len() == 1
+                && args[0].value.is_builtin(&BuiltinTypeName_::U8) =>
+        {
+            Some(T::exp(
+                ty.clone(),
+                sp(value_loc, E::Value(sp(value_loc, Value_::Bytearray(value)))),
+            ))
+        }
+        Type_::Apply(_, sp!(_, name), args) if args.is_empty() => {
+            let possibles = context.outer.get_stdlib_string_info();
+
+            for (mut str_ty, str_ctor, validator) in possibles {
+                // Expand our type first, because unification inisits they are the same.
+                type_(context, &mut str_ty);
+                if core::subtype_check(&str_ty, ty) {
```

Comment:
Should we just do a `find` here? It might be easier for me to grok? maybe not idk